### PR TITLE
Add docstrings for some functions in image, retrieval, text metrics

### LINKED ISSUE #389: Add docstrings to protected functions
## 📚 Documentation

Add docstrings to all protected functions at least tell what are the arguments for...
This can see easily implemented in multiple PRs :rabbit:

### PR DESCRIPTION
**Pull request by DevPranjal**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Partially fixes #389 

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun? Yes
Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/functional/text/bleu.py [hunk 1]
```diff
@@ -54,6 +54,19 @@ def _bleu_score_update(
     ref_len: Tensor,
     n_gram: int = 4
 ) -> Tuple[Tensor, Tensor]:
+    """
+    Updates and returns variables required to compute the BLEU score.
+
+    Args:
+        reference_corpus: An iterable of iterables of reference corpus
+        translate_corpus: An iterable of machine translated corpus
+        numerator:
```
Comment (DevPranjal) [inline torchmetrics/functional/text/bleu.py:63]: Need some help here, describing these variables. I tried to understand the context and google a few things, but could not come with concise descriptions.

Comment (Borda) [reply torchmetrics/functional/text/bleu.py]: @sameetsinghsidhu mind check :]

File: torchmetrics/functional/image/ssim.py [hunk 2]
```diff
@@ -22,6 +22,15 @@
...
 def _gaussian(kernel_size: int, sigma: float, dtype: torch.dtype, device: torch.device) -> Tensor:
+    """
+    Computes 1D gaussian kernel
+
+    Args:
+        kernel_size: size of the gaussian kernel
+        sigma: Standard deviation of the gaussian kernel
+        dtype: data type of the output tensor
+        device: device of the output tensor
+    """
```
[the remark this answers is not in the document]
Comment (SkafteNicki) [reply torchmetrics/functional/image/ssim.py:33]:
```suggestion
    Example:
        >>> _gaussian(3, 1, torch.float, 'cpu')
        tensor([[0.2741, 0.4519, 0.2741]])
    """
```

File: torchmetrics/functional/image/ssim.py [hunk 3]
```diff
@@ -30,6 +39,17 @@ def _gaussian(kernel_size: int, sigma: float, dtype: torch.dtype, device: torch.
...
+    Computes 2D gaussian kernel
+
+    Args:
+        channel: number of channels in the image
+        kernel_size: size of the gaussian kernel as a tuple (h, w)
+        sigma: Standard deviation of the gaussian kernel
+        dtype: data type of the output tensor
+        device: device of the output tensor
+    """
```
[the remark this answers is not in the document]
Comment (SkafteNicki) [reply torchmetrics/functional/image/ssim.py:51]:
```suggestion
    Example:
        >>> _gaussian_kernel(1, (5,5), (1,1), torch.float, "cpu")
        tensor([[[[0.0030, 0.0133, 0.0219, 0.0133, 0.0030],
                  [0.0133, 0.0596, 0.0983, 0.0596, 0.0133],
                  [0.0219, 0.0983, 0.1621, 0.0983, 0.0219],
                  [0.0133, 0.0596, 0.0983, 0.0596, 0.0133],
                  [0.0030, 0.0133, 0.0219, 0.0133, 0.0030]]]])
    """
```

**Review by SkafteNicki — APPROVED** [review]
Just added some example, else LGTM

File: torchmetrics/functional/image/ssim.py [hunk 4]
```diff
@@ -22,6 +22,18 @@
...
 def _gaussian(kernel_size: int, sigma: float, dtype: torch.dtype, device: torch.device) -> Tensor:
+    """
+    Computes 1D gaussian kernel
+
+    Args:
+        kernel_size: size of the gaussian kernel
+        sigma: Standard deviation of the gaussian kernel
+        dtype: data type of the output tensor
+        device: device of the output tensor
+    Example:
```
Comment (Borda) [inline torchmetrics/functional/image/ssim.py:33]:
```suggestion

    Example:
```

File: torchmetrics/functional/image/ssim.py [hunk 5]
```diff
@@ -30,6 +42,24 @@ def _gaussian(kernel_size: int, sigma: float, dtype: torch.dtype, device: torch.
...
+    """
+    Computes 2D gaussian kernel
+
+    Args:
+        channel: number of channels in the image
+        kernel_size: size of the gaussian kernel as a tuple (h, w)
+        sigma: Standard deviation of the gaussian kernel
+        dtype: data type of the output tensor
+        device: device of the output tensor
+    Example:
```
Comment (Borda) [inline torchmetrics/functional/image/ssim.py:54]:
```suggestion

    Example:
```

**Review by Borda — APPROVED** [review]

**Comment by DevPranjal** [comment]
Thanks for the changes! :)

**Comment by SkafteNicki** [comment]
@DevPranjal thanks for wanting to contribute?

**Pull request merged**